frontend/lib/api/category.ts

```diff
@@ -0,0 +1,35 @@
+import { z } from "zod";
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+import { API_BASE_URL } from "../const";
+import { Category, categorySchema } from "@/types/category";
+import { uuid } from "@/types/uuid"
+
+
+export const categoriesApi = createApi({
+  reducerPath: "categoriesApi",
+  baseQuery: fetchBaseQuery({ baseUrl: API_BASE_URL }),
+  tagTypes: ["Categories"],
+  endpoints: (builder) => ({
+    categories: builder.query<Category[], void>({
+      query: () => "categories",
+      transformResponse: (rawResponse: unknown) => {
+        return z.array(categorySchema).parse(rawResponse);
+      },
+      providesTags: [{ type: "Categories", id: "LIST" }],
```

Comment:
The docs do this for GET endpoints that return a list:

result
          ? [...result.map(({ id }) => ({ type: 'Post' as const, id })), 'Post']
          : ['Post'],